### DISCUSSION TITLE
[BUG] Get-PnPListItem doesn't always return all items

### DISCUSSION THREAD

**Discussion by hydrilllo** [pnp/powershell#2277] [General]
### Reporting an Issue or Missing Feature

I run Get-PnPListItems in a scheduled task to get all items in a list. But sometimes I only get a few items as a result.
I could reproduce the problem once in an interactive powershell session. First time I ran the command I only got the first few items.
Next time a few more and so on till I got the full list. But when I closed the powershell session opened a new one and reauthenticated I got all items immediately.

The list only contains about 50 items.

### Expected behavior 
Get all the queries items consistently

### Actual behavior 
randomly on get a subset of items

### Steps to reproduce behavior
Connect-PnpONline -Interactive -Url https://contoso.sharepoint.com/sites/Log
Get-PnpListItem -List "Log"

### What is the version of the Cmdlet module you are running?
1.11.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
hi @hydrilllo , this cmdlet works based on the logged in users/application permissions. 

If you have the permission for list items, they will be fetched. 
So, maybe there might be something else going on (caching maybe?) that maybe to causing the issue.

So, maybe in the old session, there might be a different set of permissions for list items ?
Am converting this into a discussion since it is not really an issue with cmdlet.
